trinity/utils/chains.py

```diff
@@ -151,14 +159,14 @@ def data_dir(self) -> str:
 
     @data_dir.setter
     def data_dir(self, value: str) -> None:
-        self._data_dir = os.path.abspath(value)
+        self._data_dir = PurePath(os.path.abspath(value))
 
     @property
     def database_dir(self) -> str:
         if self.sync_mode == SYNC_FULL:
-            return os.path.join(self.data_dir, DATABASE_DIR_NAME, "full")
+            return str(self.data_dir / DATABASE_DIR_NAME / "full")
```

Comment:
TIL `PurePath` objects use the division operator in place of `os.path.join`.  👍 

This syntax is going to take some time to get used to.